### ISSUE TITLE
Make the XML Schema or the Ruleset check for empty IATI Identifiers in activities

### ISSUE THREAD

**Issue by simon-20** [IATI/IATI-Schemas#500]
**Brief Description**
The Standard implicitly prohibits non-empty IATI Identifiers for activities, by saying that they MUST be composed of certain prefixes and suffixes. However, this is only in the human readable text, and not part of the Ruleset, or the XML Schema. As such, activities that have empty identifiers are currently marked as valid, and need not trigger any errors or warnings.

**Severity**
Medium 

**Issue Location**
https://iatistandard.org/en/iati-standard/203/activity-standard/iati-activities/iati-activity/iati-identifier/

**Expected Results/Behaviour**
Activities with empty identifiers should fail validation based on XML Schema, or at least the Ruleset.

**Actual Results/Behaviour**
Currently they are let through by the Validator (because it just implements the Schema and Ruleset), with the result that activities with empty identifiers can enter into the tool chain. This has caused problems for some tools.

**Comment by stevieflow** [comment]
Nice, thanks @simon-20 

Have you an example activity / file that currently illustrates this?

Thanks

**Comment by simon-20** [comment]
Yep, the activities at the end of this dataset currently have empty `<iati-identifier>` elements:

https://iatiregistry.org/dataset/ares-programme22-27

**Comment by stevieflow** [comment]
Ah yes - can see that , thanks @simon-20 

Should an activity / file fail this, it should declared as Critical imho
